examples/xlora_train/xlora_train_example_incomplete.ipynb

```diff
@@ -0,0 +1,515 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3288987d",
+   "metadata": {},
+   "source": [
+    "# Loading an X-LoRA model with PEFT, training to come!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "8274a027-6bee-4a0b-995e-30fbe883ae4d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/home/ericbuehler/miniconda3/envs/py3.12/lib/python3.12/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html\n",
+      "  from .autonotebook import tqdm as notebook_tqdm\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The token has not been saved to the git credentials helper. Pass `add_to_git_credential=True` in this function directly or `--add-to-git-credential` if using via `huggingface-cli` if you want to set the git credential as well.\n",
+      "Token is valid (permission: read).\n",
+      "Your token has been saved to /home/ericbuehler/.cache/huggingface/token\n",
+      "Login successful\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "('12.1', '2.4.0+cu121', 90100, True, True)"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import os\n",
+    "#os.environ[\"CUDA_VISIBLE_DEVICES\"] = \"0\"\n",
+    "\n",
+    "import torch\n",
+    "from transformers import AutoModelForCausalLM, AutoTokenizer\n",
+    "from transformers import BitsAndBytesConfig\n",
+    "\n",
+    "from peft import get_peft_model\n",
+    "import numpy as np\n",
+    "device = torch.device(\"cuda\")\n",
+    "\n",
+    "token = 'hf_xxxxxxxxxxxxxxxxxxxx'\n",
+    "\n",
+    "from huggingface_hub import login\n",
+    "login(token=token)\n",
+    "\n",
+    "torch.version.cuda, torch.__version__, torch.backends.cudnn.version(), torch.backends.cudnn.enabled, torch.cuda.is_available()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "3781cbdb-81c6-4d71-89a3-99b3350dd25d",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "NVIDIA GeForce RTX 4070 Laptop GPU\n"
+     ]
+    }
+   ],
+   "source": [
+    "for i in range(torch.cuda.device_count()):\n",
+    "   print(torch.cuda.get_device_properties(i).name)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "e2c18b20-b1a9-4f3e-ae84-2a551e2ed69c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def params(model):\n",
```

Comment:
This can be removed as it's not used AFAICT.

Reply:
Done!